app/controllers/messages_controller.rb

```diff
@@ -6,9 +6,7 @@ class MessagesController < ApplicationController
     controller.ensure_logged_in t("layouts.notifications.you_must_log_in_to_send_a_message")
   end
 
-  before_filter do |controller|
-    controller.ensure_authorized t("layouts.notifications.you_are_not_authorized_to_do_this")
-  end
+  ensure_can_access_person :person_id
```

Comment:
This is confusing that ensure_can_access_person expects a parameter whose value resolves to *username* and it gets passed param name called `:person_id`. With quick search I see this parameter name being referenced in only one other place so why not just rename the route parameter to avoid further confusion?